atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/persistentlock/PersistentLockIsTakenException.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright 2016 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.persistentlock;
+
+import java.util.List;
+
+public class PersistentLockIsTakenException extends Exception {
+    private final List<LockEntry> conflictingLocks;
+
+    public PersistentLockIsTakenException(List<LockEntry> conflictingLocks) {
+        this.conflictingLocks = conflictingLocks;
```

Comment:
Should probably just push the message construction into constructor as:

```
public PersistentLockIsTakenException(List<LockEntry> conflictingLocks) {
    super("Could not acquire the requested lock because it conflicts with the " +
          "following other lock(s): " + conflictingLocks);
}
```

Then we don't need to keep the conflictingLocks around (which if you were going to should be a snapshot `ImmutableSet.copyOf(conflictingLocks)` in case the version passed in isn't serializable or mutable).